e2e/journey-suites/src/webauthn-username-password.test.ts

```diff
@@ -0,0 +1,119 @@
+import { test, expect } from '@playwright/test';
+import type { CDPSession } from 'playwright';
+import { asyncEvents } from './utils/async-events.js';
+import { password, username } from './utils/demo-user.js';
+
+test.use({ browserName: 'chromium' }); // ensure CDP/WebAuthn is available
+
+test('Register and authenticate with webauthn device (username + password journey)', async ({
+  page,
+  context,
+}) => {
+  let cdp: CDPSession | undefined;
+  let authenticatorId: string | undefined;
+  let webauthnEnabled = false;
+  let recordedCredentialIds: string[] = [];
+
+  await test.step('Configure virtual authenticator', async () => {
+    cdp = await context.newCDPSession(page);
+    await cdp.send('WebAuthn.enable');
+    webauthnEnabled = true;
+
+    // A "platform" authenticator (aka internal) with UV+RK enabled is the usual default for passkeys.
+    const response = await cdp.send('WebAuthn.addVirtualAuthenticator', {
+      options: {
+        protocol: 'ctap2',
+        transport: 'internal', // platform authenticator
+        hasResidentKey: true, // allow discoverable credentials (passkeys)
+        hasUserVerification: true, // device supports UV
+        isUserVerified: true, // simulate successful UV (PIN/biometric)
+        automaticPresenceSimulation: true, // auto "touch"/presence
+      },
+    });
+    authenticatorId = response.authenticatorId;
+  });
+
+  const { navigate } = asyncEvents(page);
+
+  try {
+    // First, register a credential we can use later
+    await test.step('Navigate to registration journey', async () => {
+      await navigate('/?clientId=TEST_WebAuthn-Registration');
+      await expect(page).toHaveURL('http://localhost:5829/?clientId=TEST_WebAuthn-Registration');
+    });
+
+    await test.step('Complete primary credentials', async () => {
+      await page.getByLabel('User Name').fill(username);
+      await page.getByLabel('Password').fill(password);
+      await page.getByRole('button', { name: 'Submit' }).click();
+    });
+
+    await test.step('Register WebAuthn credential', async () => {
+      // With the virtual authenticator present and presence auto-simulated,
+      // registration will complete without any OS prompts.
+      await expect(page.getByRole('button', { name: 'Logout' })).toBeVisible();
```

Comment:
I'd like to avoid testing the test app's UI logic (e.g. using the presence of the Logout button to indirectly assert success), and focus more on testing outcomes of SDK functionality.

Reply:
we do this in almost every test suite, is this something you're suggesting we no longer do?

Reply:
Yeah, I noticed this when I was refactoring the OIDC tests, and removed it from them. I just wanted to at least not continue to do it, and refactor existing tests when we can.